lib/elsa/producer.ex

```diff
@@ -35,6 +35,7 @@ defmodule Elsa.Producer do
     case Process.whereis(registry) do
       nil ->
         {:ok, pid} = Elsa.Supervisor.start_link(endpoints: endpoints, connection: connection, producer: [topic: topic])
+        ready?(connection)
```

Comment:
does this case statement want to turn into a nested `with` block not that you have these conditions that want to execute in order and only conditionally when the previous ones succeed?